### PR TITLE
fix: extract non-number for AtomElement

### PR DESCRIPTION
**Pull request by KazuCocoa**
For https://github.com/appium/appium/issues/15683

I don't know this will help, so let keep draft this PR so far

### CODE REVIEW

File: lib/commands/web.js [hunk 1]
```diff
@@ -245,14 +245,21 @@ helpers.getAtomsElement = function getAtomsElement (wdId) {
...
+  if (_.isNull(regEl)) {
+    throw new errors.UnknownError(`Cannot convert element ID for using in WD atoms: '${el}'`);
+  }
+
+  let elementid = regEl[0];
+  if (parseInt(elementid, 10) < ELEMENT_OFFSET) {
+    log.debug(`Element with id '${elementid}' passed in for use with ` +
       `atoms, but it's out of our internal scope. Adding ${ELEMENT_OFFSET}.`);
-    el = (parseInt(el, 10) + ELEMENT_OFFSET).toString();
+    el = (parseInt(elementid, 10) + ELEMENT_OFFSET).toString();
```
Comment (mykola-mokhnach) [inline lib/commands/web.js:258]: how ELEMENT_OFFSET would work/is supposed to work with element IDs above? Could you please explain this logic?

Comment (KazuCocoa) [reply lib/commands/web.js:258]: I'm still reading why the offset is needed...
I wonder there should keep `:wdc:` for atom, but I have no environment to understand more for now...

Comment (mykola-mokhnach) [reply lib/commands/web.js:258]: yep, this is also my point. It does not make sense to add any changes there until we properly understand the purpose of the existing code (which seems redundant and sometimes useless to me) and check real usage scenarios.

Comment (jlipps) [reply lib/commands/web.js:258]: the point of ELEMENT_OFFSET was to make sure that element IDs did not clash between native and web contexts. this came from the fact that many implementations back in the day simply used numeric element ids like 1, 2, 3, etc...

Comment (KazuCocoa) [reply lib/commands/web.js:258]: ah... I see

Comment (mykola-mokhnach) [reply lib/commands/web.js:258]: Thanks for the explanation @jlipps. Do you think it makes sense to remove this constant now and use identifiers "as is"?

Do you also know what is the purpose of `webElementIds` array? Can we simply get rid of it?

**Comment by mykola-mokhnach** [comment]
I was thinking about creating a PR to change/fix this cache implementation and have it done using a common approach with LRU cache, like it is done for UIA2 or XCUITest drivers. Would that make sense @jlipps ?

File: lib/commands/web.js (hunk 1 above)
Comment (jlipps) [reply lib/commands/web.js:258]: If none of the other drivers use simple numeric IDs anymore, I think this is safe to reuse identifiers 'as is', especially if the atoms don't return numeric IDs either. But if the atoms return numeric IDs, there could still be clashes between separate contexts. Let's say a user switches back and forth to native contexts, and each time in a web context, gets the same element IDs even though they searched for different elements. Could that be a problem?

**Comment by jlipps** [comment]
Yeah, LRU cache is a good idea

**Comment by KazuCocoa** [comment]
https://github.com/appium/appium-xcuitest-driver/pull/1324 worked with `:wdc:1628151649325` by Safari.

**Pull request closed without merging**